### DISCUSSION TITLE
middleware vs server/middleware for user auth

### DISCUSSION THREAD

**Discussion by AnkushSarkar10** [nuxt/nuxt#22332] [Questions] [answered]
whats the difference between middleware directory and server/middleware directory? If i am to check for user auth before they can access a rout, which one should I use and why?

**Comment by AloisSeckar** [comment, marked as answer]
`/middleware` runs within Vue part of the app and it is executed before navigation to matching route
`/server/middleware` runs within Nitro server part of the app and is executed before ANY server api request

Checking user auth definitely belongs to `/middleware`

**Reply by sumasagrp** [reply]
to extend to this server middleware will also run on all initial Vue routes ( if they were accessed directly ).  This caught me out before. Hopefully this can be added to docs. 
source:  https://github.com/nuxt/nuxt/issues/24531#issuecomment-1833527571
